### ISSUE TITLE
Missing `CreateEmailOptions` type in exports

### ISSUE THREAD

**Issue by MatteoGauthier** [resend/resend-node#287]
Hello ! I've just noticed that Type `CreateEmailOptions` has been removed recently, I'd like to know why, otherwise I have an alternative to this type :

```ts
type CreateEmailOptions = Parameters<typeof resend.emails.create>[0]
```

Have a nice day

**Comment by yingw787** [comment]
^ +1

**Comment by JClackett** [comment]
yes pleaseeeeee, creating a generic helper function is quite a common use case for these types

**Comment by taylor-lindores-reeves** [comment]
Yes, why has it been removed?

**Comment by bukinoshita** [comment]
Thanks for reporting this, let's definitely re-add. Can one work on PR on this?

**Comment by QuentinFrc** [comment]
#122

**Comment by wscourge** [comment]
Just expose all I/O types related to the public API.

**Comment by JClackett** [comment]
```ts
// src/index.ts
export { Resend } from './resend';
export { ErrorResponse  } from './interfaces';
export { CreateEmailOptions } from "./emails/interfaces/create-email-options.interface";
```

is all we need, do I really need to create a PR for this?

**Comment by amirfakhrullah** [comment]
Hi, can we merge #305 ? And maybe export other types as well (`CreateEmailRequestOptions` , `CreateEmailResponse`)

**Comment by MartinXPN** [comment]
We still can't use the latest version of the node client (without weird workarounds) and are forced to stay on `resend@2.0.0` just because of this.
It would be great to expose `CreateEmailOptions` and `CreateEmailResponse`.
Do you think it would be possible to expose these types along with adding the `statusCode` key to `ErrorResponse` (it's currently missing)? @bukinoshita @zenorocha

**Comment by MartinXPN** [comment]
Here is a workaround for the latest version (`3.2.0` as of now):

```ts
import {Resend, ErrorResponse as ErrorResponseType} from 'resend';

type CreateEmailOptions = Parameters<typeof Resend.prototype.emails.send>[0];
type CreateEmailResponse = Awaited<ReturnType<typeof Resend.prototype.emails.send>>;
type ErrorResponse = ErrorResponseType & {statusCode: number};
type ResendResponse = CreateEmailResponse | ErrorResponse;
```

**Comment by ahrbil** [comment]
For anyone using pnpm, you can path `resend` package and export all the types you need.
Run `pnpm path resend@<version>`
Update `dist/index.d.ts` file in the extracted folder to export the types you want.
Then run `pnpm patch-commit <path to the extracted directory>`
At the end restart your ts server.

**Comment by robert-bo-davis** [comment]
#396 appears to resolve this

**Comment by vcapretz** [comment]
all (most?) of the types should be exported now, please open a new issue if you're missing some specific type 😄
